### DISCUSSION TITLE
Should `as_tensor_variable` work with any variables that are accepted by `numpy.asarray`?

### DISCUSSION THREAD

**Discussion by ricardoV94** [aesara-devs/aesara#912] [Ideas]
Here is one example with pandas `RangeIndex` objects:

```python
from pandas import RangeIndex
from numpy import asarray
from aesara.tensor import as_tensor_variable

x = RangeIndex(1, 5)
asarray(x)  # Fine
as_tensor_variable(x)  # NotImplementedError
```

**Comment by brandonwillard** [comment]
I think you're referring to something that might involve [NumPy's generalized `__array_function__` interface](https://numpy.org/doc/stable/reference/arrays.classes.html#numpy.class.__array_function__) with that example.  This is related to https://github.com/aesara-devs/aesara/discussions/802 and abstracted away from NumPy in https://github.com/aesara-devs/aesara/issues/729.
